clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -306,6 +312,10 @@ public boolean partitionAssignmentNeeded() {
         return this.needsPartitionAssignment;
     }
 
+    /**
+     * Change the assignment to the specified partitions returned from the coordinator,
+     * note this is different from {@link #assign(List)} which directly set the assignment from user inputs
+     */
     public void changePartitionAssignment(Collection<TopicPartition> assignments) {
```

Comment:
Not sure I see the harm in referring to the coordinator (we already do so in the comments), but that works too.